dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1460,7 +1462,40 @@ class CompileTest {
       case DeviceOperatingSystem.fuchsia:
         throw Exception('Unsupported option for Fuchsia devices');
       case DeviceOperatingSystem.macos:
-        throw Exception('Unsupported option for macOS devices');
+        unawaited(stderr.flush());
+        options.insert(0, 'macos');
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final String buildDirectoryPath = path.join(
+          cwd,
+          'build',
+          'macos',
+          'Build',
+          'Products',
+          'Release',
+        );
+        final String? appBundlePath =
+            _findIosAppInBuildDirectory(buildDirectoryPath);
+        if (appBundlePath == null) {
+          throw 'Failed to find app bundle in $buildDirectoryPath';
+        }
+        final Directory appBundle = Directory(appBundlePath);
+        final Archive archive = Archive();
+        await appBundle
+            .list(recursive: true)
+            .where((FileSystemEntity e) => e.runtimeType == File)
+            .map<ArchiveFile>((FileSystemEntity e) {
+              assert(e.runtimeType == File);
+              final File file = e as File;
+              final Uint8List content = file.readAsBytesSync();
```

Comment:
Use [`File.statSync`](https://api.flutter.dev/flutter/dart-io/FileSystemEntity/statSync.html) here, then grab the [`size`](https://api.flutter.dev/flutter/dart-io/FileStat/size.html) from that.

Reply:
@cbracken instead of archiving the compilation output. I've chosen record the size of the build output in the number of bytes. Please resolve as you see fit:
https://github.com/flutter/flutter/blob/d70df45883b810910e8b90a9e5aec1f750869b7b/dev/devicelab/lib/tasks/perf_tests.dart#L1465-L1471